RBV801/movie-search-test2
# Add AI Search Explanation Panel

This pull request implements an AI search explanation panel to enhance search transparency and user understanding.

Implementation Details:
- Added AISearchExplanation component
- Created searchAnalysis utility
- Integrated with main App component
- Added AI analysis state management

Testing Instructions:
1. Run application locally
2. Test various search types
3. Verify AI panel displays
4. Check credit counter
5. Confirm panel updates
